x-pack/plugins/cases/public/components/case_view/case_view_metrics.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import { EuiFlexGroup, EuiFlexItem, EuiLoadingSpinner, EuiPanel } from '@elastic/eui';
+import { CaseMetrics } from '../../../common/ui';
+import {
+  ASSOCIATED_HOSTS_METRIC,
+  ASSOCIATED_USERS_METRIC,
+  TOTAL_ALERTS_METRIC,
+  TOTAL_CONNECTORS_METRIC,
+} from './translations';
+
+const MetricValue = styled(EuiFlexItem)`
+  font-size: ${({ theme }) => theme.eui.euiSizeL};
+  font-weight: bold;
+`;
+
+export interface CaseViewMetricsProps {
+  metrics: CaseMetrics | null;
+  isLoading: boolean;
+}
+
+type MetricItems = Array<{ title: string; value: number }>;
+
+const useMetricItems = (metrics: CaseMetrics | null): MetricItems => {
+  const { alerts, connectors } = metrics ?? {};
+  const totalConnectors = connectors?.length;
+  const alertsCount = alerts?.count;
+  const totalAlertUsers = alerts?.users?.total;
+  const totalAlertHosts = alerts?.hosts?.total;
+
+  const metricItems = useMemo<MetricItems>(() => {
+    const items = [];
+    if (alertsCount != null) {
+      items.push({ title: TOTAL_ALERTS_METRIC, value: alertsCount });
+    }
+    if (totalAlertUsers != null) {
+      items.push({ title: ASSOCIATED_USERS_METRIC, value: totalAlertUsers });
+    }
+    if (totalAlertHosts != null) {
+      items.push({ title: ASSOCIATED_HOSTS_METRIC, value: totalAlertHosts });
+    }
+    if (totalConnectors != null) {
+      items.push({ title: TOTAL_CONNECTORS_METRIC, value: totalConnectors });
+    }
```

Comment:
```suggestion
     return items.filter(({ value }) => value != null)
```

where `const items = [{ title: TOTAL_ALERTS_METRIC, value: alertsCount }, { title: ASSOCIATED_USERS_METRIC, value: totalAlertUsers }, ...]`